hibernate-core/src/main/java/org/hibernate/dialect/DerbyDialect.java

```diff
@@ -105,78 +107,107 @@ public DerbyDialect() {
 
 	public DerbyDialect(DatabaseVersion version) {
 		super(version);
-		registerDerbyKeywords();
 	}
 
 	public DerbyDialect(DialectResolutionInfo info) {
 		super(info);
-		registerDerbyKeywords();
 	}
 
 	@Override
-	protected String columnType(int jdbcTypeCode) {
-		if ( jdbcTypeCode == BOOLEAN && getVersion().isBefore( 10, 7 ) ) {
-			return "smallint";
-		}
-
-		switch (jdbcTypeCode) {
+	protected String columnType(int sqlTypeCode) {
+		switch ( sqlTypeCode ) {
+			case BOOLEAN:
+				return getVersion().isBefore( 10, 7 ) ? "smallint" : super.columnType( sqlTypeCode );
 			case TINYINT:
 				//no tinyint
 				return "smallint";
-
 			case NUMERIC:
 				// HHH-12827: map them both to the same type to avoid problems with schema update
 				// Note that 31 is the maximum precision Derby supports
-				return super.columnType(DECIMAL);
-
+				return columnType( DECIMAL );
 			case VARBINARY:
 				return "varchar($l) for bit data";
-
+			case LONGVARBINARY:
+			case LONG32VARBINARY:
+				return "long varchar for bit data";
+			case NCHAR:
+				return columnType( CHAR );
+			case NVARCHAR:
+				return columnType( VARCHAR );
+			case LONGNVARCHAR:
+				return columnType( LONGVARCHAR );
+			case LONGVARCHAR:
+			case LONG32VARCHAR:
+				return "long varchar";
```

Comment:
Like, are you copy/pasting this code, which was previously defined on the superclass (DRY) down to the subclasses?

Please don't do that.